components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/indexing/indexer/CustomAPIIndexer.java

```diff
@@ -69,6 +82,20 @@ public IndexDocument getIndexedDocument(AsyncIndexer.File2Index fileData) throws
                 resource.setProperty(CUSTOM_API_INDEXER_PROPERTY, "true");
                 registry.put(resourcePath, resource);
             }
+
+            if (storeVisibility != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("API at " + resourcePath + "did not have property : " + APIConstants.STORE_VIEW_ROLES
+                            + ", hence adding the values for that API resource.");
+                }
+                if (storeVisibility.equals("public")) {
+                    resource.setProperty(APIConstants.STORE_VIEW_ROLES, "null");
+                } else if (storeVisibility.equals("restricted")){
```

Comment:
We can use constants for "public" , "null" and "restricted" in this block I think. Shall we do that?